src/applications/oscilloscope.py

```diff
@@ -1,117 +1,247 @@
+parser.add_argument('-rmw', '--rolling-mean-window', metavar='milliseconds', default=400, type=float,
+                    help='''Sets the displayed rolling mean window size, w, (in milliseconds).
+                    Actual value will set as the ceil(rmw/aut) * aut.''')
+parser.add_argument('-srm', '--show-rolling-mean', default='Υ',
+                    help='''Show (Y) or hide (N) the rolling mean line.''')
+parser.add_argument('-srt', '--show-reading-text', default='Υ',
+                    help='''Show (Y) or hide (N) the reading text. 
+                    If rolling mean is enabled, rolling mean, rolling standard
+                    deviation (STD) and expected STD (from Poison distribution) 
+                    text will also be displayed.''')
+
+parser.add_argument('--console', action='store_true',
                     help='Run as console app -- just the figure without buttons.')
 args = parser.parse_args()
 
 
-
 class ScopeFigure:
 
-    def __init__(self, width=50, fig = None, ax = None):
-        if ax == None:
+    def __init__(self, width: int = 50, rolling_mean_window_size: int = 20,
+                 show_rolling_mean: bool = True, show_reading_text: bool = True,
+                 fig: plt.Figure = None, ax: plt.Axes = None):
+        """
+        Initializes the ScopeFigure.
+
+        Parameters
+        ----------
+        width: int, default 50
+            Number of points to display in the figure.
+        rolling_mean_window_size: int, default 20
+            Size of the rolling mean window.
+        show_rolling_mean: bool, default True
+            Whether to display the rolling mean line in the figure.
+        show_reading_text: bool, default True
+             Whether to display current reading value as text.
+        fig : matplotlib.figure.Figure
+            Figure to plot the data in.
+        ax : matplotlib.axes.Axes
+            Axis to plot the data in.
+        """
+
+        # setting up the figure and main axis
+        if ax is None:
             fig, ax = plt.subplots()
             self.fig = fig
             self.ax = ax
         else:
             self.fig = fig
             self.ax = ax
+        self.fig.set_layout_engine('compressed')
 
+        # display settings
+        self.show_rolling_mean = show_rolling_mean
+        self.show_reading_text = show_reading_text
+
+        # setting up primary data plot
         self.ydata = collections.deque(np.zeros(width))
         self.line, = self.ax.plot(self.ydata)
-        self.ax.set_ylabel('counts / sec')
-        self.ax.ticklabel_format(style='sci',scilimits=(-3,4),axis='y')
+        self.line: plt.Line2D
+
+        # Setting up axis preferences
+        # TODO: Put all axis preferences to .mplstyle and load through terminal or gui.
+        self.ax.set_ylabel('Reading (counts / sec)')
+        self.ax.ticklabel_format(style='sci', scilimits=(-3, 4), axis='y')
+        self.ax.set_xlabel('Acquisition Number')
```

Comment:
It would be nice if we could format the x-axis so it displays values of time..... Any other labels we could use here? "Time bin"? or "Time (arb. units)" until somebody does the work to convert the values to actual time values.

Reply:
I am planning on implementing a different method on collecting data with the new device interface class, which will always include time stamps. For now, maybe we can multiply the index number with the acquisition period?

Reply:
looking into it, I realized that the ScopeFigure does not know anything about the update interval, so we should probably leave it that way for now. Updated it to "Time bin (arb. units)"